algebird-core/src/main/scala/com/twitter/algebird/BloomFilter.scala

```diff
@@ -150,6 +154,8 @@ case class BFItem(item: String, hashes: BFHash, width: Int) extends BF {
 
   def +(other: String) = this ++ BFItem(other, hashes, width)
```

Comment:
in ++ should we optimize the case where `other.item == item` and not allocate? I guess that's rare.